cli/src/main.rs

```diff
@@ -15,25 +18,159 @@ fn to_commit_hash(s: &str) -> Result<CommitHash> {
+            simperby_node.create_agenda().await?;
+        }
+        Commands::Vote { commit } => {
+            let mut simperby_node = initialize(config, &path).await?;
+            simperby_node
+                .vote(
+                    serde_spb::from_str(&commit)
+                        .map_err(|_| eyre!("invalid agenda commit hash to vote on"))?,
+                )
+                .await?;
+        }
+        Commands::Veto { commit } => {
+            let mut simperby_node = initialize(config, &path).await?;
+            if commit.is_none() {
+                simperby_node.veto_round().await?;
+            } else {
+                simperby_node
+                    .veto_block(
+                        serde_spb::from_str(&commit.expect("commit is not none"))
+                            .map_err(|_| eyre!("invalid block commit hash to veto on"))?,
+                    )
+                    .await?;
+            }
+        }
+        Commands::Consensus { show } => {
+            if show {
+                // TODO: show the status of the consensus instead of making a progress.
+            } else {
+                let mut simperby_node = initialize(config, &path).await?;
+                simperby_node.progress_for_consensus().await?;
+            }
+        }
         Commands::Git => todo!(),
-        Commands::Clean { .. } => todo!(),
-        Commands::Create(CreateCommands::Agenda) => todo!(),
-        Commands::Create(CreateCommands::Block) => todo!(),
-        Commands::Vote { .. } => todo!(),
-        Commands::Veto { .. } => todo!(),
         Commands::Show { commit } => show(config, &path, commit).await?,
-        Commands::Consensus { show: _ } => todo!(),
         Commands::Network => todo!(),
-        Commands::Serve => todo!(),
-        Commands::Update => todo!(),
-        Commands::Broadcast => todo!(),
-        Commands::Chat { .. } => todo!(),
+        Commands::Serve => {
+            serve(config, &path).await?;
+        }
+        Commands::Update => {
+            let mut simperby_node = initialize(config, &path).await?;
+            simperby_node.fetch().await?;
+        }
+        Commands::Broadcast => {
+            let mut simperby_node = initialize(config, &path).await?;
+            simperby_node.broadcast().await?;
+        }
+        Commands::Chat { .. } => todo!("chat is not implemented yet"),
+        Commands::Sign(SignCommands::TxDelegate {
+            delegatee,
+            governance,
+            target_height,
+        }) => {
+            let delegation_transaction_data = DelegationTransactionData {
+                delegator: config.public_key,
+                delegatee: serde_spb::from_str(&delegatee)
+                    .map_err(|_| eyre!("invalid delegatee for a delegation transaction"))?,
+                governance,
+                block_height: target_height,
+            };
+            println!(
+                "{:?}",
```

Comment:
Considering the use of this command, I think it's better to print json

Reply:
Agree, fixed.